### PR TITLE
Dev working

### PR DESCRIPTION
**Pull request by weitliu**

### CODE REVIEW

**Comment by bkj** [comment]
Can you add some comments explaining the changes you've made?

File: compare.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+./bin/gtf
+sleep 1
+./bin/ffa
+sleep 1
+echo "please ignore the above output. Comparing the from ./output_graphtv.txt and ./output_ffa.txt"
+diff output_graphtv.txt output_ffa.txt
```
Comment (bkj) [inline compare.sh]: `compare.sh` is dead code?

File: ffagtf/min.cpp [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    graph_tv(Y, n, m, edges1, edges2, lambda1, 0.0);
+    soft_thresh(Y, lambda2, n);
+    t2 = clock();
+
+    cout << "time is " << ((float)t2 - (float)t1) / CLOCKS_PER_SEC << endl;
+    ofstream out( "./output_ffa.txt" );
+    for(int i = 0; i < n; i++) out << Y[i] << endl;
+
+  return 0;
+}
```
Comment (bkj) [inline ffagtf/min.cpp]: `min.cpp` never gets compiled or run?  If that's the case, it should be removed.

File: graphtv/gtf.cpp [hunk 3]
```diff
@@ -7,48 +7,45 @@
 #include <map>
 #include <set>
 #include <sstream>
+#include "graph.h"
```
Comment (bkj) [inline graphtv/gtf.cpp]: importing header twice

File: graphtv/maxflow.cpp [hunk 4]
```diff
@@ -500,6 +500,13 @@ Graph::flowtype Graph::maxflow()
 	}
 
 	delete nodeptr_block;
+
+	printf("\ntr_cap in this run is: \n");
+  for (i=node_block->ScanFirst(); i; i=node_block->ScanNext()) {
+    printf("%f \n",i->tr_cap);
+  }
+
+
```
Comment (bkj) [inline graphtv/maxflow.cpp:509]: Does this `maxflow` function ever get called?  Dead code should be removed

**Review by bkj — COMMENTED** [review]
Thanks for the PR.  There's some dead code that I point out in the comments below -- would be good to remove that.

Running `./run.sh` here fails the validation in the same way that the current `master` branch does:
```
generating graph
make-graph.py: wrote _data/{e,n}
running various versions
SuperGraph: None partition
SuperGraph: 1 partitions in 0.000061s
===========================
validate gtf ffa
nonzero entries:
[[-3.13 -3.13]
 [ 0.   -0.77]
 [ 0.    0.66]
 [-3.13 -3.13]
 [ 0.   -0.48]
 [ 0.    0.71]
 [ 0.   -0.78]
 [ 0.    0.2 ]
 [ 0.   -0.39]]
FAIL
0.000000 != -0.770000
0.000000 != 0.660000
0.000000 != -0.480000
0.000000 != 0.710000
0.000000 != -0.780000
0.000000 != 0.200000
0.000000 != -0.390000
===========================
validate gtf reference
nonzero entries:
[[-3.13 -3.13]
 [ 0.   -0.77]
 [ 0.    0.66]
 [-3.13 -3.13]
 [ 0.   -0.48]
 [ 0.    0.71]
 [ 0.   -0.78]
 [ 0.    0.2 ]
 [ 0.   -0.39]]
FAIL
0.000000 != -0.770000
0.000000 != 0.660000
0.000000 != -0.480000
0.000000 != 0.710000
0.000000 != -0.780000
0.000000 != 0.200000
0.000000 != -0.390000
===========================
validate ffa reference
nonzero entries:
[[-3.13 -3.13]
 [-0.77 -0.77]
 [ 0.66  0.66]
 [-3.13 -3.13]
 [-0.48 -0.48]
 [ 0.71  0.71]
 [-0.78 -0.78]
 [ 0.2   0.2 ]
 [-0.39 -0.39]]
PASS
```
That is, `ffa` and `reference` agree, but `graphtv` does not.  It would be good to know why this is the case so we make sure we're understanding what's going on. Though remember -- the most critical thing is that the Gunrock GPU implementation produces the same results as the reference implementation.

~ Ben

**Comment by weitliu** [comment]
can you introduce what this function is: erdos_renyi_graph? it says a binomial graph but I cannot search any information on this. I guess the problem is the graph itself, not the algorithm since my previous graph generation is always consistent and on the NY taxi dataset the graphtv is consistent with the python results too.

File: compare.sh (hunk 1 above)
Comment (weitliu) [reply compare.sh]: removed

File: ffagtf/min.cpp (hunk 2 above)
Comment (weitliu) [reply ffagtf/min.cpp]: removed

File: graphtv/gtf.cpp (hunk 3 above)
Comment (weitliu) [reply graphtv/gtf.cpp]: line removed.

File: graphtv/maxflow.cpp (hunk 4 above)
Comment (weitliu) [reply graphtv/maxflow.cpp:509]: this maxflow is needed, as the parametric maxflow is calling the last two functions in this file.

**Comment by weitliu** [comment]
I also added the graph option directed in the make-graph

**Pull request merged**